src/cmdline.rs

```diff
@@ -406,6 +433,28 @@ fn parse_download(matches: &ArgMatches) -> Result<Config> {
     }))
 }
 
+fn parse_list_stream(matches: &ArgMatches) -> Result<Config> {
+    let stream_base_url = if matches.is_present("stream-base-url") {
+        Some(
+            Url::parse(
+                matches
+                    .value_of("stream-base-url")
+                    .expect("stream-base-url missing"),
```

Comment:
I think this can be avoided by adjusting the logic like:
```
let stream_base_url = match matches.value_of("stream-base-url") {
  Some(u) => { ... },
  None => None,
}
```